### ISSUE TITLE
Trouble running this

### ISSUE THREAD

**Issue by cmrockwell** [dkowis/Cucumber-JVM-Ruby-Example#4]
This projects looks like a nice way to get started with jruby for cucumber-jvm. But I'm having trouble getting started. I have pasted the mvn output showing the errors I'm getting.  I'm using Maven 3.1.1 and Java 1.7.0_51. I've tried it with jruby 1.7.4 and 1.7.20. The output  below is using ruby-1.7.4. Any suggestions would be appreciated...

*mvn test -Pinitial-bundle
2379591019:Cucumber-JVM-Ruby-Example cmrockwe$ mvn test -Pinitial-bundle
[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building Examples: Ruby Stack 1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ ruby-stack-example ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] skip non existing resourceDirectory /Users/cmrockwe/Documents/a3workspace/Cucumber-JVM-Ruby-Example/src/main/resources
[INFO] 
[INFO] --- maven-compiler-plugin:2.5.1:compile (default-compile) @ ruby-stack-example ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-antrun-plugin:1.6:run (default) @ ruby-stack-example ---
[INFO] Executing tasks

main:
     [echo] Installing bundler gem
     [java] Successfully installed bundler-1.9.9
     [java] 1 gem installed
     [echo] Initializing bundle
     [java] Gemfile already exists at /Users/cmrockwe/Documents/a3workspace/Cucumber-JVM-Ruby-Example/Gemfile
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 23.196s
[INFO] Finished at: Sun May 24 16:56:06 EDT 2015
[INFO] Final Memory: 8M/245M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:1.6:run (default) on project ruby-stack-example: An Ant BuildException has occured: Java returned: 1 -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
2379591019:Cucumber-JVM-Ruby-Example cmrockwe$ 

AND

*mvn clean test
2379591019:Cucumber-JVM-Ruby-Example cmrockwe$ mvn clean test
[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building Examples: Ruby Stack 1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ ruby-stack-example ---
[INFO] Deleting /Users/cmrockwe/Documents/a3workspace/Cucumber-JVM-Ruby-Example/target
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ ruby-stack-example ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] skip non existing resourceDirectory /Users/cmrockwe/Documents/a3workspace/Cucumber-JVM-Ruby-Example/src/main/resources
[INFO] 
[INFO] --- maven-compiler-plugin:2.5.1:compile (default-compile) @ ruby-stack-example ---
[WARNING] File encoding has not been set, using platform encoding UTF-8, i.e. build is platform dependent!
[INFO] Compiling 1 source file to /Users/cmrockwe/Documents/a3workspace/Cucumber-JVM-Ruby-Example/target/classes
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ ruby-stack-example ---
[WARNING] File encoding has not been set, using platform encoding UTF-8, i.e. build is platform dependent!
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] Copying 3 resources
[INFO] 
[INFO] --- maven-compiler-plugin:2.5.1:testCompile (default-testCompile) @ ruby-stack-example ---
[WARNING] File encoding has not been set, using platform encoding UTF-8, i.e. build is platform dependent!
[INFO] Compiling 1 source file to /Users/cmrockwe/Documents/a3workspace/Cucumber-JVM-Ruby-Example/target/test-classes
[INFO] 
[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ ruby-stack-example ---
[INFO] Surefire report directory: /Users/cmrockwe/Documents/a3workspace/Cucumber-JVM-Ruby-Example/target/surefire-reports

-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running com.rackspace.example.stackTest
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.313 sec <<< FAILURE!
initializationError(com.rackspace.example.stackTest)  Time elapsed: 0.005 sec  <<< ERROR!
cucumber.runtime.CucumberException: java.lang.reflect.InvocationTargetException
	at cucumber.runtime.io.ResourceLoaderReflections.newInstance(ResourceLoaderReflections.java:81)
	at cucumber.runtime.io.ResourceLoaderReflections.instantiateSubclasses(ResourceLoaderReflections.java:57)
	at cucumber.runtime.Runtime.loadBackends(Runtime.java:78)
	at cucumber.runtime.Runtime.<init>(Runtime.java:52)
	at cucumber.api.junit.Cucumber.<init>(Cucumber.java:59)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
	at org.junit.internal.builders.AnnotatedBuilder.buildRunner(AnnotatedBuilder.java:29)
	at org.junit.internal.builders.AnnotatedBuilder.runnerForClass(AnnotatedBuilder.java:21)
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
	at org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass(AllDefaultPossibilitiesBuilder.java:26)
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
	at org.junit.internal.requests.ClassRequest.getRunner(ClassRequest.java:26)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:250)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
	at cucumber.runtime.io.ResourceLoaderReflections.newInstance(ResourceLoaderReflections.java:75)
	... 26 more
Caused by: cucumber.runtime.CucumberException: Invalid RUBY_VERSION: jruby-1.7.4
	at cucumber.runtime.jruby.JRubyBackend.<init>(JRubyBackend.java:67)
	... 31 more

Results :

Tests in error: 
  initializationError(com.rackspace.example.stackTest): java.lang.reflect.InvocationTargetException

Tests run: 1, Failures: 0, Errors: 1, Skipped: 0

**Comment by dkowis** [comment]
It's been a while since I updated this guy. I'll take a look tomorrow! 

On May 24, 2015 4:06:13 PM CDT, Chris Rockwell <notifications@github.com> wrote:
>This projects looks like a nice way to get started with jruby for
>cucumber-jvm. But I'm having trouble getting started. I have pasted the
>mvn output showing the errors I'm getting.  I'm using Maven 3.1.1 and
>Java 1.7.0_51. I've tried it with jruby 1.7.4 and 1.7.20. The output 
>below is using ruby-1.7.4. Any suggestions would be appreciated...
>
>
>*mvn test -Pinitial-bundle
>2379591019:Cucumber-JVM-Ruby-Example cmrockwe$ mvn test
>-Pinitial-bundle
>[INFO] Scanning for projects...
>[INFO]                                                                 
>       
>[INFO]
>------------------------------------------------------------------------
>[INFO] Building Examples: Ruby Stack 1.0-SNAPSHOT
>[INFO]
>------------------------------------------------------------------------
>[INFO] 
>[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @
>ruby-stack-example ---
>[WARNING] Using platform encoding (UTF-8 actually) to copy filtered
>resources, i.e. build is platform dependent!
>[INFO] skip non existing resourceDirectory
>/Users/cmrockwe/Documents/a3workspace/Cucumber-JVM-Ruby-Example/src/main/resources
>[INFO] 
>[INFO] --- maven-compiler-plugin:2.5.1:compile (default-compile) @
>ruby-stack-example ---
>[INFO] Nothing to compile - all classes are up to date
>[INFO] 
>[INFO] --- maven-antrun-plugin:1.6:run (default) @ ruby-stack-example
>---
>[INFO] Executing tasks
>
>main:
>     [echo] Installing bundler gem
>     [java] Successfully installed bundler-1.9.9
>     [java] 1 gem installed
>     [echo] Initializing bundle
>[java] Gemfile already exists at
>/Users/cmrockwe/Documents/a3workspace/Cucumber-JVM-Ruby-Example/Gemfile
>[INFO]
>------------------------------------------------------------------------
>[INFO] BUILD FAILURE
>[INFO]
>------------------------------------------------------------------------
>[INFO] Total time: 23.196s
>[INFO] Finished at: Sun May 24 16:56:06 EDT 2015
>[INFO] Final Memory: 8M/245M
>[INFO]
>------------------------------------------------------------------------
>[ERROR] Failed to execute goal
>org.apache.maven.plugins:maven-antrun-plugin:1.6:run (default) on
>project ruby-stack-example: An Ant BuildException has occured: Java
>returned: 1 -> [Help 1]
>[ERROR] 
>[ERROR] To see the full stack trace of the errors, re-run Maven with
>the -e switch.
>[ERROR] Re-run Maven using the -X switch to enable full debug logging.
>[ERROR] 
>[ERROR] For more information about the errors and possible solutions,
>please read the following articles:
>[ERROR] [Help 1]
>http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
>2379591019:Cucumber-JVM-Ruby-Example cmrockwe$ 
>
>
>AND
>
>*mvn clean test
>2379591019:Cucumber-JVM-Ruby-Example cmrockwe$ mvn clean test
>[INFO] Scanning for projects...
>[INFO]                                                                 
>       
>[INFO]
>------------------------------------------------------------------------
>[INFO] Building Examples: Ruby Stack 1.0-SNAPSHOT
>[INFO]
>------------------------------------------------------------------------
>[INFO] 
>[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @
>ruby-stack-example ---
>[INFO] Deleting
>/Users/cmrockwe/Documents/a3workspace/Cucumber-JVM-Ruby-Example/target
>[INFO] 
>[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @
>ruby-stack-example ---
>[WARNING] Using platform encoding (UTF-8 actually) to copy filtered
>resources, i.e. build is platform dependent!
>[INFO] skip non existing resourceDirectory
>/Users/cmrockwe/Documents/a3workspace/Cucumber-JVM-Ruby-Example/src/main/resources
>[INFO] 
>[INFO] --- maven-compiler-plugin:2.5.1:compile (default-compile) @
>ruby-stack-example ---
>[WARNING] File encoding has not been set, using platform encoding
>UTF-8, i.e. build is platform dependent!
>[INFO] Compiling 1 source file to
>/Users/cmrockwe/Documents/a3workspace/Cucumber-JVM-Ruby-Example/target/classes
>[INFO] 
>[INFO] --- maven-resources-plugin:2.6:testResources
>(default-testResources) @ ruby-stack-example ---
>[WARNING] File encoding has not been set, using platform encoding
>UTF-8, i.e. build is platform dependent!
>[WARNING] Using platform encoding (UTF-8 actually) to copy filtered
>resources, i.e. build is platform dependent!
>[INFO] Copying 3 resources
>[INFO] 
>[INFO] --- maven-compiler-plugin:2.5.1:testCompile
>(default-testCompile) @ ruby-stack-example ---
>[WARNING] File encoding has not been set, using platform encoding
>UTF-8, i.e. build is platform dependent!
>[INFO] Compiling 1 source file to
>/Users/cmrockwe/Documents/a3workspace/Cucumber-JVM-Ruby-Example/target/test-classes
>[INFO] 
>[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @
>ruby-stack-example ---
>[INFO] Surefire report directory:
>/Users/cmrockwe/Documents/a3workspace/Cucumber-JVM-Ruby-Example/target/surefire-reports
>
>-------------------------------------------------------
> T E S T S
>-------------------------------------------------------
>Running com.rackspace.example.stackTest
>Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.313
>sec <<< FAILURE!
>initializationError(com.rackspace.example.stackTest)  Time elapsed:
>0.005 sec  <<< ERROR!
>cucumber.runtime.CucumberException:
>java.lang.reflect.InvocationTargetException
>	at
>cucumber.runtime.io.ResourceLoaderReflections.newInstance(ResourceLoaderReflections.java:81)
>	at
>cucumber.runtime.io.ResourceLoaderReflections.instantiateSubclasses(ResourceLoaderReflections.java:57)
>	at cucumber.runtime.Runtime.loadBackends(Runtime.java:78)
>	at cucumber.runtime.Runtime.<init>(Runtime.java:52)
>	at cucumber.api.junit.Cucumber.<init>(Cucumber.java:59)
>	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native
>Method)
>	at
>sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
>	at
>sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
>	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
>	at
>org.junit.internal.builders.AnnotatedBuilder.buildRunner(AnnotatedBuilder.java:29)
>	at
>org.junit.internal.builders.AnnotatedBuilder.runnerForClass(AnnotatedBuilder.java:21)
>	at
>org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
>	at
>org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass(AllDefaultPossibilitiesBuilder.java:26)
>	at
>org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
>	at
>org.junit.internal.requests.ClassRequest.getRunner(ClassRequest.java:26)
>	at
>org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:250)
>	at
>org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
>	at
>org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
>	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>	at
>sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
>	at
>sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
>	at java.lang.reflect.Method.invoke(Method.java:606)
>	at
>org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
>	at
>org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
>	at
>org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
>	at
>org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
>	at
>org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
>Caused by: java.lang.reflect.InvocationTargetException
>	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native
>Method)
>	at
>sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
>	at
>sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
>	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
>	at
>cucumber.runtime.io.ResourceLoaderReflections.newInstance(ResourceLoaderReflections.java:75)
>	... 26 more
>Caused by: cucumber.runtime.CucumberException: Invalid RUBY_VERSION:
>jruby-1.7.4
>	at cucumber.runtime.jruby.JRubyBackend.<init>(JRubyBackend.java:67)
>	... 31 more
>
>
>Results :
>
>Tests in error: 
>initializationError(com.rackspace.example.stackTest):
>java.lang.reflect.InvocationTargetException
>
>Tests run: 1, Failures: 0, Errors: 1, Skipped: 0
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/dkowis/Cucumber-JVM-Ruby-Example/issues/4

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by dkowis** [comment]
Two problems:
1. You don't run the `initial-bundle` profile unless you don't have a gemfile. It will install the bundler gem and create one for you. Even the failure doesn't actually mean anything if you already have a gem file. It could probably be done better, but maven's flexibility isn't so great.

2. I had to include the `RSpec::Matchers` in the rack_stack_steps.rb file, because the methods weren't being recognized. A google search led me to this one, I've pushed that up to master. 

I'm probably going to update all the things to the latest version of cucumber-jvm to make sure it all still works.
